File: integrationtest/src/main/java/com/sequenceiq/it/IntegrationTestContext.java
package com.sequenceiq.it;

import java.util.HashMap;
import java.util.Map;

public class IntegrationTestContext {
    public static final String AUTH_TOKEN = "AUTH_TOKEN";

    private Map<String, Object> contextParameters = new HashMap<>();
    private Map<String, Object> cleanUpParameters = new HashMap<>();

    public IntegrationTestContext() {
    }

    public IntegrationTestContext(Map<String, Object> contextParameters) {
        this.contextParameters = contextParameters;
    }

    public String getContextParam(String paramKey) {
        return getContextParam(paramKey, String.class);
    }

    public <T> T getContextParam(String paramKey, Class<T> clazz) {
        Object val = contextParameters.get(paramKey);
        if (val == null || clazz.isInstance(val)) {
            return clazz.cast(val);
        } else {
            throw new IllegalArgumentException("Param value is not type of " + clazz);
        }
    }

    public void putContextParam(String paramKey, Object paramValue) {
        putContextParam(paramKey, paramValue, false);
    }

    public void putContextParam(String paramKey, Object paramValue, boolean cleanUp) {
        contextParameters.put(paramKey, paramValue);
        if (cleanUp) {
            putCleanUpParam(paramKey, paramValue);
        }
    }

    public void putCleanUpParam(String paramKey, Object paramValue) {
        cleanUpParameters.put(paramKey, paramValue);
    }

    public String getCleanUpParameter(String key) {
        return getCleanUpParameter(key, String.class);
    }

    public <T> T getCleanUpParameter(String key, Class<T> clazz) {
        Object val = cleanUpParameters.get(key);
        if (val == null || clazz.isInstance(val)) {
            return clazz.cast(val);
        } else {
            throw new IllegalArgumentException("Param value is not type of " + clazz);
        }
    }
}

File: integration-test/src/main/java/com/sequenceiq/it/IntegrationTestContext.kt
package com.sequenceiq.it

import java.util.HashMap

class IntegrationTestContext {

    private val contextParameters = HashMap<String, Any>()
    private val cleanUpParameters = HashMap<String, Any>()

    constructor() {
    }

    constructor(contextParameters: MutableMap<String, Any>) {
        this.contextParameters = contextParameters
    }

    fun getContextParam(paramKey: String): String {
        return getContextParam(paramKey, String::class.java)
    }

    fun <T> getContextParam(paramKey: String, clazz: Class<T>): T {
        val `val` = contextParameters[paramKey]
        if (`val` == null || clazz.isInstance(`val`)) {
            return clazz.cast(`val`)
        } else {
            throw IllegalArgumentException("Param value is not type of " + clazz)
        }
    }

    @JvmOverloads fun putContextParam(paramKey: String, paramValue: Any, cleanUp: Boolean = false) {
        contextParameters.put(paramKey, paramValue)
        if (cleanUp) {
            putCleanUpParam(paramKey, paramValue)
        }
    }

    fun putCleanUpParam(paramKey: String, paramValue: Any) {
        cleanUpParameters.put(paramKey, paramValue)
    }

    fun getCleanUpParameter(key: String): String {
        return getCleanUpParameter(key, String::class.java)
    }

    fun <T> getCleanUpParameter(key: String, clazz: Class<T>): T {
        val `val` = cleanUpParameters[key]
        if (`val` == null || clazz.isInstance(`val`)) {
            return clazz.cast(`val`)
        } else {
            throw IllegalArgumentException("Param value is not type of " + clazz)
        }
    }

    companion object {
        val IDENTITY_URL = "IDENTITY_URL"
        val AUTH_USER = "AUTH_USER"
        val AUTH_PASSWORD = "AUTH_PASSWORD"
    }
}
